lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+     * Removes a font family.
+     *
+     * Removes a font family from an array of font families. If the font family to remove contains font face 
+     * definitions, the associated assets are deleted.
+     *
+     * @param array $font_families The current array of font families.
+     * @param array $font_to_uninstall The font family to remove from the font families array.
+     * @return array The updated array of font families.
+     */
+    function remove_font_family ( $font_families, $font_to_uninstall ) {
+        $new_font_families = array();
+        foreach ( $font_families as $font_family ) {
+            if ( $font_family['slug'] !== $font_to_uninstall['slug'] ) {
+                $new_font_families[] = $font_family;
+            } else {
+                if ( isset ( $font_family['fontFace'] ) ){
+                    foreach( $font_family['fontFace'] as $font_face ) {
+                        $this->delete_font_face_assets( $font_face );
+                    }
+                }
+            }
+        }
+        return $new_font_families;
+    }
+
+    /**
+     * Installs new fonts.
+     *
+     * Takes a request containing new fonts to install, downloads their assets, and adds them to the fonts library.
+     *
+     * @param WP_REST_Request $request The request object containing the new fonts to install in the request parameters.
+     * @return WP_REST_Response|WP_Error The updated fonts library post content.
+     */
+    function install_fonts ( $request ) {
+        // Get existing font families
+        $post = $this->query_fonts_library();
+        if ( is_wp_error( $post ) ) {
+            return $post;
+        }
+        $post_content = $post->post_content;
+        $library = json_decode( $post_content, true );
+        $font_families = $library['fontFamilies'];
+
+        // Get new fonts to install
+        $fonts_to_install = $request->get_param('fontFamilies');
+        if ( is_string( $fonts_to_install ) ) {
+            // If we are receiving form data (as we to upload local fonts), the font families are encoded as a string.
+            $fonts_to_install = json_decode( $fonts_to_install, true );
+        }
+
+        if ( empty ( $fonts_to_install ) ) {
+            return new WP_Error( 'no_fonts_to_install', __( 'No fonts to install' ), array( 'status' => 400 ) );
+        }
+
+        //  Get uploaded files (used when installing local fonts)
+        $files = $request->get_file_params();
+
+        // Download or move the new fonts assets to the fonts folder
+        $new_fonts = $this->download_or_move_fonts( $fonts_to_install, $files );
+
+        if ( ! empty ( $new_fonts )  ){
+            // Updates the fonts library with the new fonts succesfully downloaded
+            $new_library_fonts = $this->merge_fonts( $font_families, $new_fonts );
+
+            // Sanitizes the fonts library using WP_Theme_JSON
+            $sanitized_font_families = $this->sanitize_font_families( $new_library_fonts );
```

Comment:
`$sanitized_font_families` is not used and can be removed.
```suggestion
            $this->sanitize_font_families( $new_library_fonts );
```